packages/perspective-viewer-d3fc/src/js/tooltip/tooltip.js

```diff
@@ -92,9 +103,21 @@ function showTooltip(containerNode, barNode, tooltipDiv) {
         .duration(200)
         .style("opacity", 0.9);
 
+    if (centered) left = centerTip(tooltipDiv, containerRect);
```

Comment:
Why does this happen after setting `left` and `top` for the div where as centring the top happens before? Also, this one calls a function called `centerTip` which only centres horizontally.

Reply:
Yeah. On your second point, I'm going to go back and change that. At the time, I took out the final vertical centering because in the only case where it's being used it just frankly looked better. This was because the tooltip would be drawn just beneath the treemap's label for the rectangle. Now that in subsequent iterations the treemap labels are themselves completely centered vertically, this no longer makes sense, as the tooltip cuts the label in half. So yes. This shall be changed post-haste.

As to point 1, in all other cases, we want the tooltip to spawn from the middle of the bar or point horizontally, but also the very top. Only in the treemap case do we want the tooltip to spawn in the middle, vertically speaking. Best way to see the pre-existing paradigm is looking at one of the bar charts.